### PR TITLE
ui: Notifications re-organization/re-style

### PR DESCRIPTION
**Pull request by johncowen**
This PR re-organizes our global UI notifications slightly:

1. Moves where they appear up to the `<App />` component, this solves all sorts of craziness that we've had since the initial v2 rebuild of the UI.
2. Instead of a `<Notification />` wrapping component to move whatever you use for a notification up to where they need to appear (via `ember-cli-flash`), we now use a `{{notification}}` modifier now we have modifiers.
3. Global notifications/flashes are no longer special styles of their own. You just use the `{{notification}}` modifier to hoist whatever component/element you want up to the top of the page. This means we can re-use our existing `<Notice />` component for all our global UI notifications (this is the user visible change here)

All [notifications now look like this](https://consul-ui-staging-1rjnxowt5-hashicorp.vercel.app/ui/docs/modifiers/notification) i.e. the same as our [`Notice` component](https://consul-ui-staging-1rjnxowt5-hashicorp.vercel.app/ui/docs/components/notice)

This is only ~80% towards where I want this to end up.

1. I found that we use `data-notification` for test selecting, this shows how old this stuff is. It should be `data-test-notification`, and ideally a pageobject. This code would have originated from the first few weeks of the build. For the moment I've centralized this in one place in the app (the modifier) and left it as something I can do in a follow up PR as it will mean a bunch of test selector/page-object wrangling and this PR was already getting a little big.
2. Our models that still don't use our new componentized approach for CRUD have had their `Consul::----::Notifications` components moved up to `<Hashicorp-Consul />`. This was just easier for the moment, especially considering the next large chore is probably moving all these to our new style CRUD (getting rid of the old style mixins that are left)

There might be a bit more to add here, so I'll either do that later or inline if I spot anything useful to add. I also split the commits up a little to vaguely describe groups of changes there.

Further notes:

1. We half-used an existing `{{style}}` modifier which will be useful for us in other places in the UI, but I wanted to be able to delay style modifications easily for animation styling purposes. So I took it and modified it slightly to allow me to do that. I might clean it all up a little further down the line.
2. `<AppView />` is now pretty much just a glorified partial. At some point soon I'll probably update this component to be called `<Page />` and upgrade it to glimmer/native-named-slots soon. This was one of the large benefits and objectives of doing this overall.

### CODE REVIEW

**Review by evrowe — CHANGES_REQUESTED** [review]
Mostly just left a few questions; found one typo that needs changing but otherwise this looks good to go

File: ui/packages/consul-ui/app/components/hashicorp-consul/index.hbs
Comment (johncowen) [reply ui/packages/consul-ui/app/components/hashicorp-consul/index.hbs:70]: Yeah I think I was just being lazy cos this is soon to be going and I did the copy/pasta bit by bit. This is literally to get me through the next month or so. I suppose if it stayed like this and we changed it up to use a dynamic `{{component}}` could error if I ever did `flash.model = "service"`?

Happy to change up if you want, I'm easy, I'm guessing from where we are right now, the ACL ones will end up in a release (I'm getting rid of the nspace one in https://github.com/hashicorp/consul/pull/11633 after the rebase 😬 )

Let's chat about this one later.

File: ui/packages/consul-ui/app/modifiers/notification.mdx
Comment (johncowen) [reply ui/packages/consul-ui/app/modifiers/notification.mdx:5]: Yup, that was the was the realisation I had also 😸 

There's a little gotcha tho 🤫 don't tell anyone 😄 You lose any event handlers or reactiveness when the DOM element is 'hoisted' as it essentially stringifies the DOM and then reinserts it into the DOM via `outerHTML`. I only sleep well at night as I know we don't need this functionality (yet at least 😬 ). I'm sure there'll be a way around that if/when we do though.

We also have `<Portal />`s (a 3rd party dep) which **does** keep all the events/reactivity, but unfortunately we can't use this here as the extra feature we need in this case is for the original DOM to stick around on the page even when the component where it originally took the outerHTML from has gone from the page. Its a fun problem I'd eventually like to fix properly, we can chat a bit more about this later also if you are interested more.

File: ui/packages/consul-ui/app/modifiers/notification.mdx
Comment (johncowen) [reply ui/packages/consul-ui/app/modifiers/notification.mdx:31]:
```suggestion
Currently this is backed by `ember-cli-flash` and the named options are
```

File: ui/packages/consul-ui/app/modifiers/style.js
Comment (johncowen) [inline ui/packages/consul-ui/app/modifiers/style.js:52]: FYI: theres already a different plan to re-assess this in an un-blinkered state.

**Review by johncowen — COMMENTED** [review]
I added a GH suggestion here (which I'll accept/commit now) for the typo, plus some answers to your queries. So I think its good to go, unless we want to do the `{{component}}` thing also, which is fine by me if we do. Lets touch on this PR later.

File: ui/packages/consul-ui/app/modifiers/notification.js
Comment (evrowe) [reply ui/packages/consul-ui/app/modifiers/notification.js:15]: Got it, seems fine to leave it as-is then.

File: ui/packages/consul-ui/app/components/hashicorp-consul/index.hbs
Comment (evrowe) [reply ui/packages/consul-ui/app/components/hashicorp-consul/index.hbs:70]: If this is temporary then yeah let's not worry about it. If we figure out that it needs to remain longer term, then we can circle back around and deal with it later.

**Review by evrowe — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/508854.

**Pull request merged**
